devenv/scripts/deploy_private_e2e_tester.sh

```diff
@@ -0,0 +1,41 @@
+CLUSTER_RESOURCE_GROUP=$(cat state/cluster-info.json | jq '.ClusterResourceGroup' | tr -d '"')
+CLUSTER_NAME=$(cat state/cluster-info.json | jq '.ClusterName' | tr -d '"')
+
+export IMAGE=$(cat state/e2e-image-tag.txt)
+envsubst < e2e-tester.yaml > state/e2e-tester-formatted.yaml
+
+echo "deleting any existing e2e job..."
+az aks command invoke --resource-group $CLUSTER_RESOURCE_GROUP --name $CLUSTER_NAME --command "kubectl delete jobs app-routing-operator-e2e -n kube-system"
+
+set -e
+az aks command invoke --resource-group $CLUSTER_RESOURCE_GROUP --name $CLUSTER_NAME --command "kubectl apply -f e2e-tester-formatted.yaml" --file state/e2e-tester-formatted.yaml
+
+RESULT="null"
+while [ "$RESULT" = "null" ]
+do
+  STATUS=$(az aks command invoke --resource-group $CLUSTER_RESOURCE_GROUP --name $CLUSTER_NAME --command 'kubectl get pods --selector=app=app-routing-operator-e2e --output="jsonpath={.items[0].status.containerStatuses[?(@.name==\"tester\")].state}" -n kube-system' -o json | jq ".logs" | tr -d '\\')
+  STATUSNOQUOTES=${STATUS#"\""}
+  STATUSNOQUOTES=${STATUSNOQUOTES%"\""}
+
+  RESULT=$(echo $STATUSNOQUOTES | jq '.terminated')
+  echo "test status is currently $STATUSNOQUOTES"
+
+  sleep 5
+
+done
+
+echo "exited loop with status $STATUSNOQUOTES"
+
+FINALSTATUS=$(echo $RESULT | jq ".exitCode")
+
+echo "Test finished, echoing test pod logs..."
```

Comment:
why are we using this script instead of using go tests like the other e2e tests?

Reply:
this uses the same e2e test that is used for public clusters. This script simply runs that same go code on the private cluster by deploying a pod on the private cluster that then calls make e2e.
We can't do it via go code because we can't access private clusters via a Kubernetes client on a machine outside the private cluster's VM.
Plus, this way, we have one suite that runs for both private and public clusters, so if changes are made/tests are added to this one suite, those changes propagate to both kinds of clusters.